rosidl_actions/rosidl_actions/__init__.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+from rosidl_cmake import expand_template
+from rosidl_cmake import get_newest_modification_time
+from rosidl_cmake import read_generator_arguments
+from rosidl_parser import parse_action_file
+
+
+def generate_msg_and_srv(generator_arguments_file):
+    args = read_generator_arguments(generator_arguments_file)
+
+    for ros_interface_file in args['ros_interface_files']:
+        extension = os.path.splitext(ros_interface_file)[1]
+        subfolder = os.path.basename(os.path.dirname(ros_interface_file))
+        if extension == '.action':
+            services, message = parse_action_file(args['package_name'], ros_interface_file)
+
+            # create folder if necessary
+            try:
+                os.makedirs(os.path.join(args['output_dir'], subfolder))
+            except FileExistsError:
```

Comment:
@sloretz why not check for file existence instead e.g `if not os.path.exists(path): ...`? I'm not particularly fond of using exceptions for flow control unless it's strictly necessary.

Reply:
Fixed in 37a485f4c8d69ec0f8f3eced2e09d3dbc67da4c8